From b28d09c1a4f72c64382c57a714012b04fa02585b Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sun, 13 Nov 2022 20:43:17 +0100
Subject: [PATCH 11/14] Failing to create facedata shouldn't be fatal

(new version of phh patch for november 2022 release android 11)

Some Pie vendors create it on their own, so SELinux would deny that
Also not all devices have face unlock anyway

See https://github.com/phhusson/treble_experimentations/issues/1119
---
 vold_prepare_subdirs.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vold_prepare_subdirs.cpp b/vold_prepare_subdirs.cpp
index d624d73..60056f3 100644
--- a/vold_prepare_subdirs.cpp
+++ b/vold_prepare_subdirs.cpp
@@ -164,7 +164,7 @@ static bool prepare_subdirs(const std::string& volume_uuid, int user_id, int fla
             }
             auto facedata_path = vendor_de_path + "/facedata";
             if (!prepare_dir(sehandle, 0700, AID_SYSTEM, AID_SYSTEM, facedata_path)) {
-                return false;
+                LOG(ERROR) << "Failed preparing folder for de facedata";
             }
         }
         if (flags & android::os::IVold::STORAGE_FLAG_CE) {
@@ -187,7 +187,7 @@ static bool prepare_subdirs(const std::string& volume_uuid, int user_id, int fla
             auto vendor_ce_path = android::vold::BuildDataVendorCePath(user_id);
             auto facedata_path = vendor_ce_path + "/facedata";
             if (!prepare_dir(sehandle, 0700, AID_SYSTEM, AID_SYSTEM, facedata_path)) {
-                return false;
+                LOG(ERROR) << "Failed preparing folder for de facedata";
             }
         }
     }
-- 
2.25.1

